Add Client::getLocation() helper

clue opened the pull request.

clue commented:
See the individual commits for details why this is work-in-progress.

Any input is welcome!

floriansimon1 commented:
I  think I've implemented that feature here : https://github.com/floriansimon1/php-soap-react/commit/913d3faf65f28a808126d85f400b7ec52a75b8d7

It's in the last lines of my ClientEncoder changes ;)

clue commented:
> I've implemented that feature here : floriansimon1@913d3fa

This is true :) Despite some differences in the implementation, I'd rather keep these two features separate, hence this PR.

I've just updated the PR and resolved the outstanding issues. Would you care to review this? :)

floriansimon1 commented:
I did not know that individual webservice function could have different locations.

> I've just updated the PR and resolved the outstanding issues. Would you care to review this? :)
I just did. What I'd add is a way to get the WSDL location, because when you use a webservice with a single location for each function, you don't want to specify a function name, because it would add unintuitive boilerplate to your code. Moreover, having to specify a function name introduces an unnecessary source of error. What if you mistype the function name ? Or the function name does not exist anymore ?

Otherwise, I have to say I like your implementation more than my own. Thank you for your work !

clue commented:
> What I'd add is a way to get the WSDL location, because when you use a webservice with a single location for each function, you don't want to specify a function name, because it would add unintuitive boilerplate to your code. Moreover, having to specify a function name introduces an unnecessary source of error. What if you mistype the function name ? Or the function name does not exist anymore ?

Thanks for bringing this up, I think this is a valid concern :+1: 

I've just updated the implementation to also accept a function index.

As such, you can now always call it like this for your use case:

```php
$location = $client->getLocation(0);
```
